### DISCUSSION TITLE
Projeto do curso de  Segurança em APIs ASP.NET com JWT e Bearer Authentication

### DISCUSSION THREAD

**Discussion by sandrosousamiketec** [balta-io/forum#552] [Tira Dúvidas] [answered]
Olá Balta, eu fiz esse curso e sobre essa arquitetura Baseada em UseCase já tive contato em node, mas foi novo pra mim isso em .net, sempre fui acostumado a utilizar a abordagem de design de software como o (DDD),  quando sei que devo utilizar entre um deles?

**Comment by andrebaltieri** [comment]
Bom dia, @Sandro-Sousa, como vai?

É uma abordagem parecida em certos pontos, visto que o DDD trabalha mais no centro, os Use Cases acabam complementando o domínio, extendendo as regras.

No fim é mais uma questão de gosto e necessidade mesmo :) 
Não tem um "jeito certo" que funciona para tudo.

**Reply by sandrosousamiketec** [reply, marked as answer]
Tudo Beleza, Agraceço @andrebaltieri fico muito Agradecido pela explicação me tirou essa dúvida, vou estudar mais sobre e implementar isso em meus projetos.
